File: Daily Challanges/2026-07-July/2026-07-10.py
def exact_change(amount):
    coins = [1, 5, 10, 25]
    dp = [0] * (amount + 1)
    dp[0] = 1

    for coin in coins:
        for i in range(coin, amount + 1):
            dp[i] += dp[i - coin]

    return dp[amount]


print(exact_change(3))
print(exact_change(9))
print(exact_change(17))
print(exact_change(39))
print(exact_change(61))
print(exact_change(99))

File: Daily Challanges/2026-07-July/2026-07-10.js
function exactChange(amount) {
  const coins = [1, 5, 10, 25];
  const dp = new Array(amount + 1).fill(0);
  dp[0] = 1;

  for (const coin of coins) {
    for (let i = coin; i <= amount; i++) {
      dp[i] += dp[i - coin];
    }
  }

  return dp[amount];
}

console.log(exactChange(3));
console.log(exactChange(9));
console.log(exactChange(17));
console.log(exactChange(39));
console.log(exactChange(61));
console.log(exactChange(99));
